homeassistant/components/mqtt/mixins.py

```diff
@@ -237,6 +259,25 @@ async def __call__(
         """Define setup_entities type."""
 
 
+async def async_get_platform_config_from_yaml(
+    hass: HomeAssistant, domain: str, schema: vol.Schema
+) -> list[ConfigType]:
+    """Return a list of validated configurations for the platform read from configuration.yaml."""
+
+    config_yaml: ConfigType = hass.data.get(DATA_MQTT_CONFIG, {})
+    if not (platform_configs := config_yaml.get(domain)):
+        return []
+    errors = []
+    try:
+        config = [schema(config) for config in platform_configs]
+    except (ValueError, vol.MultipleInvalid) as exc:
+        errors.append(exc)
+    if errors:
+        raise vol.MultipleInvalid(errors)
```

Comment:
With https://github.com/home-assistant/core/pull/71995 this can be simplified:
```suggestion
    config = platform_configs
```

Reply:
I think we can fix that when #71995 is to be merged. The `ValueError` could already be taken out since we only want to catch schema errors.